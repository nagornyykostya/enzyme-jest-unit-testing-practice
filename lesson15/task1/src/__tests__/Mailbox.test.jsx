import React from "react";
import { shallow } from "enzyme";
import Mailbox from "../Mailbox.jsx";

describe(`<Mailbox>`, () => {
  it("should display number of unread messages", () => {
    const props = {
      unreadMessages: [1, 2, 3],
    };
    const wrappedComponent = shallow(<Mailbox {...props} />);
    expect(wrappedComponent.find(".mailbox__count").exists()
    ).toBeTruthy();
  });
});

describe(`<Mailbox>`, () => {
  it("should not display number of unread messages", () => {
    const props = {
      unreadMessages: [],
    };
    const wrappedComponent = shallow(<Mailbox {...props} />);
    expect(wrappedComponent.find(".mailbox__count").exists()
    ).toBeFalsy();
  });
});
