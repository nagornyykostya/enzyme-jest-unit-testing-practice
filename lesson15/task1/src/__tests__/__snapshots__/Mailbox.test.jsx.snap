// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mailbox> should should display number of unread messages 1`] = `ShallowWrapper {}`;
